### PR TITLE
Extend Doc.__init__ with additional annotation

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Mostly copying from `spacy.tests.util.get_doc`, add additional kwargs to `Doc.__init__` to initialize the most common doc/token values.

### Types of change

Enhancement.

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
This doesn't do `sent_starts`. It wouldn't be too hard to add if desired.

File: spacy/tokens/doc.pyx [hunk 1]
```diff
@@ -158,17 +158,38 @@ cdef class Doc:
             raise ValueError(Errors.E046.format(name=name))
         return Underscore.doc_extensions.pop(name)
 
-    def __init__(self, Vocab vocab, words=None, spaces=None, user_data=None):
+    def __init__(
+        self,
+        Vocab vocab,
+        words=None,
+        spaces=None,
+        user_data=None,
```
Comment (ines) [inline spacy/tokens/doc.pyx:166]: I wonder if we could make the `user_data` keyword-only 🤔 I doubt it's used much in user code and we haven't really been using it either. So I think it'd be safe?

**Review by ines — APPROVED** [review]

File: spacy/tokens/doc.pyx (hunk 1 above)
Comment (adrianeboyd) [reply spacy/tokens/doc.pyx:166]: I didn't want to change it, but it wasn't even in the docs.

Comment (ines) [reply spacy/tokens/doc.pyx:166]: If it was undocumented, then let's just go or it and make it keyword-only!

**Comment by honnibal** [comment]
I think the `sent_starts` is probably a good addition, yeah. Otherwise, looks good!

**Pull request merged**